vortex-array/src/lib.rs

```diff
@@ -240,6 +245,38 @@ pub trait ArrayTrait:
         self.accept(&mut visitor).unwrap();
         visitor.0
     }
+
+    fn as_null_array(&self) -> Option<&dyn NullArrayTrait> {
```

Comment:
ergonomically, we should also have `_unchecked` versions of each of these that do the unwrap for you.

or maybe we should do the unwrap by default and have an explicit `_opt` that returns the optional

Reply:
I'm not sure where to do the dtype check. In practice, you can cast into a type-specific trait if the encoding supports it, regardless of the current DType.